app/src/main/kotlin/ooo/akito/webmon/ui/createentry/CreateEntryActivity.kt

```diff
@@ -406,6 +414,86 @@ class CreateEntryActivity : AppCompatActivity() {
     }
   }
 
+  val CUSTOM_PREF_NAME = "Entry_data"
+
+  object PreferenceHelper {
+
+    val NAME = "NAME"
+    val URL = "URL"
+    val isDNSChecked = "DNSCheck"
+    val isLaissezFaireChecked = "LaissezCheck"
+
+    fun defaultPreference(context: Context): SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+
+    fun customPreference(context: Context, name: String): SharedPreferences = context.getSharedPreferences(name, Context.MODE_PRIVATE)
+
+    inline fun SharedPreferences.editMe(operation: (SharedPreferences.Editor) -> Unit) {
+      val editMe = edit()
+      operation(editMe)
+      editMe.apply()
+    }
+
+    var SharedPreferences.saveURL
+      get() = getString(URL, "")
+      set(value) {
+        editMe {
+          it.putString(URL, value)
+        }
+      }
+
+    var SharedPreferences.saveIsDNSChecked
+      get() = getBoolean(isDNSChecked, false)
+      set(value) {
+        editMe {
+          it.putBoolean(isDNSChecked, value)
+        }
+      }
+
+    var SharedPreferences.saveName
+      get() = getString(NAME, "")
+      set(value) {
+        editMe {
+          it.putString(NAME, value)
+        }
+      }
+
+    var SharedPreferences.saveIsLaissezFaireChecked
+      get() = getBoolean(isLaissezFaireChecked, false)
+      set(value) {
+        editMe {
+          it.putBoolean(isLaissezFaireChecked, value)
+        }
+      }
+  }
```

Comment:
Could you please put this into a separate file? The Activity files are always so huge and I don't want to bloat them additionally, when unnecessary.

Reply:
Sure, so you want me to create a new class in the ui.createentry folder?